lib/matplotlib/cbook.py

```diff
@@ -2385,7 +2384,7 @@ def pts_to_midstep(x, *args):
     # do normalization
     vertices = _step_validation(x, *args)
     # create the output array
-    steps = ma.zeros((vertices.shape[0], 2 * len(x)), np.float)
+    steps = np.zeros((vertices.shape[0], 2 * len(x)), np.float)
```

Comment:
`pts_to_prestep` was already using `np.zeros` so if there was a bug it was already there, if any; but in fact `Line2D` handles replacing masked arrays by nan-filled arrays (see `Line2D.recache`).